addons/pos_mercury/static/src/js/pos_mercury.js

```diff
@@ -125,6 +125,10 @@ pos_model.Paymentline = pos_model.Paymentline.extend({
         if (this.mercury_card_number) {
             this.name = this.mercury_card_brand + " (****" + this.mercury_card_number + ")";
         }
+    },
+    is_done: function () {
+        var res = _paylineproto.is_done.apply(this);
+        return res && !this.mercury_swipe_pending;
```

Comment:
shouldn't we keep using get_payment_status == 'done' and, calling 'set_payment_status' with correct status when we are setting 'mercury_swipe_pending' ?

Reply:
Indeed, that will be less code changes.

Reply:
I tried your suggestion but it seems to be not a good idea because pos_mercury is not yet adapted to behave as `payment-terminal` as it is not using the set_payment_status and get_payment_status interface. And if I call set_payment_status('done') when mercury_swipe_pending is set to false, the original implementation of get_total_paid in the Order model will still count the pending mercury payment because pending mercury payment doesn't have payment_status.

I'd say we keep it this way, that mercury does not rely on get_payment_status and set_payment_status. Larger refactor is needed if we introduce these two functions in pos_mercury.